### PR TITLE
Fix virtual OSD elements and OSD backwards compability bug

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #3114 

Also fixes backwards compatibility in OSD.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by sugaarK — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]
Thanks!

**Pull request merged**
